allennlp/modules/seq2seq_encoders/bidirectional_language_model_transformer.py

```diff
@@ -259,16 +265,34 @@ def forward(self, token_embeddings: torch.Tensor, mask: torch.Tensor) -> torch.T
         forward_mask, backward_mask = self.get_attention_masks(mask.int())
         token_embeddings = self._position(token_embeddings)
         token_embeddings = self._dropout(token_embeddings)
-        forward_output = self._forward_transformer(token_embeddings, forward_mask)
-        backward_output = self._backward_transformer(token_embeddings, backward_mask)
 
-        if self._return_all_layers:
-            to_return = []
-            for forward, backward in zip(forward_output, backward_output):
-                to_return.append(torch.cat([forward, backward], -1))
-            return to_return
+        if self.go_forward:
+            forward_output = self._forward_transformer(token_embeddings, forward_mask)
+        if self.go_backward:
+            backward_output = self._backward_transformer(token_embeddings, backward_mask)
+
+        if self.go_forward and self.go_backward:
+            if self._return_all_layers:
+                to_return = []
+                for forward, backward in zip(forward_output, backward_output):
+                    to_return.append(torch.cat([forward, backward], -1))
+                return to_return
+            return torch.cat([forward_output, backward_output], -1)
+        elif self.go_forward and not self.go_backward:
+            if self._return_all_layers:
+                to_return = []
+                for forward in forward_output:
+                    to_return.append(forward)
+                return to_return
+            return forward_output
+        elif self.go_backward and not self.go_forward:
```

Comment:
Nit: I think the redundancy in this block can be reduced by doing something like:
```
            if self._return_all_layers:
                to_return = []
                for forward, backward in zip(forward_output, backward_output):
                    layers = []
                    if self.go_forward:
                        layers.append(forward)
                    if self.go_backward:
                        layers.append(backward)
                    to_return.append(torch.cat(layers, -1))
                return to_return
            layers = []
            if self.go_forward:
                layers.append(forward_output)
            if self.go_backward:
                layers.append(backward_output)
            return torch.cat(layers, -1)
```